lisa/tools/git.py

```diff
@@ -153,3 +155,39 @@ def init_submodules(self, cwd: pathlib.PurePath) -> None:
             expected_exit_code=0,
             expected_exit_code_failure_message="error on init submodules.",
         )
+
+    def get_tag(
+        self,
+        cwd: pathlib.PurePath,
+        sort_by: str = "v:refname",
+        contains: str = "",
+        return_last: bool = True,
+    ) -> str:
+        sort_arg = ""
+        contains_arg = ""
+        if sort_by:
+            # git tag exposes various sort options, apply them if present
+            # default is sort by version, ascending
+            sort_arg = f"--sort={sort_by}"
+        if contains:
+            # git tag allows you to filter by a commit id, apply it is present.
+            contains_arg = f"--contains {contains}"
+
+        git_cmd = f"--no-pager tag {sort_arg} {contains_arg}"
+        tags = self.run(
+            git_cmd,
+            cwd=cwd,
+            expected_exit_code=0,
+            expected_exit_code_failure_message=(
+                "git tag failed to fetch tags, "
+                "check sort and commit arguments are correct"
+            ),
+        ).stdout.splitlines()
+        assert_that(len(tags)).described_as(
+            "Error: this repo contained no tags"
```

Comment:
Because the method supports filter, so the reason may be like "no tag is found".